force-app/main/default/lwc/serviceDeliveryRow/serviceDeliveryRow.js

```diff
@@ -147,19 +151,69 @@ export default class ServiceDeliveryRow extends LightningElement {
             });
     }
 
+    handleGetServicesForProgramEngagement(programEngagementId) {
+        this.isError = false;
+
+        getServicesByProgramEngagementId({ programEngagementId: programEngagementId })
+            .then(result => {
+                if (result) {
+                    this._filteredServices = result[SERVICES].slice(0);
+                } else {
+                    this.isError = true;
+                }
+
+                this.enableServiceInputWithOptions();
+            })
+            .catch(err => {
+                this.rowError = handleError(err, false);
+            });
+    }
+
     handleInputChange(event) {
-        if (event.target.fieldName === this.fields.contact.fieldApiName) {
+        if (
```

Comment:
This method is getting pretty long, I'm wondering if it'd be worth splitting it up a bit. Thinking maybe something like handleContactChange() then we can return early if it is not the contact field?
Then another one like handleProgramEngagementChange() were we can again return early if it's not the PE field.

Reply:
I made the change and it looks so clean. Thank you for the suggestion